src/util/strconv.h

```diff
@@ -127,4 +129,61 @@ static char *rm_strdupcase(const char *s, size_t len) {
   return ret;
 }
 
-#endif
\ No newline at end of file
+static char *rm_strdupcase_utf8(const char *s, size_t len) {
+  // TODO: set locale depending on the index language?
+  setlocale(LC_ALL, "en_US.UTF-8");
+
+  // Allocate memory for the destination string
+  char *ret = rm_strndup(s, len);
+  if (!ret) {
+    return NULL;
+  }
+
+  char *dst = ret;
+  char *src = dst;
+  mbstate_t state;
+  memset(&state, 0, sizeof(state));
+  wchar_t wc;
+  size_t len_wc;
+
+  while (*src) {
+    len_wc = mbrtowc(&wc, src, MB_CUR_MAX, &state);
+    if (len_wc == (size_t)-1 || len_wc == (size_t)-2) {
+      // Handle invalid multi-byte sequence
+      src++;
+      continue;
+    }
+
+    // Unescape
+    if (wc == L'\\' && (iswpunct(*(src + len_wc)) || iswspace(*(src + len_wc)))) {
+      src++;
+      continue;
+    }
+
+    wc = towlower(wc);
+    len_wc = wcrtomb(dst, wc, &state);
+    if (len_wc == (size_t)-1) {
+      // Handle conversion error
+      src++;
+      continue;
+    }
+
+    dst += len_wc;
+    src += len_wc;
+  }
+
+  *dst = '\0';
+
+  return ret;
+}
+
+// strndup + lowercase in one pass!
+static char *rm_strdupcase(const char *s, size_t len) {
+  if (setlocale(LC_ALL, "en_US.UTF-8") != NULL) {
```

Comment:
maybe add a config command to set the locale?
Don't think this should keep overriding the existing local.
It might cause modules to fight on the locale the process will use.

Also in the documentation it says modifying the local in one thread while another is using the locale is undefined behaviour.
Not sure if we should use a global function to determine what a local function in the context of a thread will choose to do.

Reply:
Consider using `newlocale` instead.